app/services/stomp-websocket.js

```diff
@@ -27,28 +29,30 @@ export default Service.extend({
     return this.get('settings.queryStompResponseChannel');
   }),
 
-  makeStompMessageHandler(stompClient, successHandler, context) {
+  makeStompMessageHandler(stompClient, context) {
     return payload => {
       let message = JSON.parse(payload.body);
       if (!isEqual(message.type, ACK_TYPE)) {
         if (isEqual(message.type, COMPLETE_TYPE)) {
-          stompClient.disconnect();
+          this.get('querier').cancel();
         }
-        successHandler(JSON.parse(message.content), context);
+        context.get('queryManager').addSegment(context.get('result'), JSON.parse(message.content));
```

Comment:
We should not expect context to have a queryManager. We should inject it here ourselves. It is also weird that this knows there will be a result variable in context.

Should we pass in a messageHandler like we do a successHandler and an errorHandler?